src/main/webapp/app/iris/base-chatbot/iris-base-chatbot.component.ts

```diff
@@ -0,0 +1,371 @@
+                    height: '0',
+                    transform: 'scale(0)',
+                }),
+                group([
+                    animate(
+                        '0.3s ease-in-out',
+                        style({
+                            height: '*',
+                        }),
+                    ),
+                    animate(
+                        '0.3s 0.1s cubic-bezier(.2,1.22,.64,1)',
+                        style({
+                            transform: 'scale(1)',
+                        }),
+                    ),
+                ]),
+            ]),
+        ]),
+    ],
+})
+export class IrisBaseChatbotComponent implements OnInit, OnDestroy, AfterViewInit {
+    // Icons
+    faTrash = faTrash;
+    faCircle = faCircle;
+    faPaperPlane = faPaperPlane;
+    faExpand = faExpand;
+    faXmark = faXmark;
+    faArrowDown = faArrowDown;
+    faCircleInfo = faCircleInfo;
+    faCompress = faCompress;
+    faThumbsUp = faThumbsUp;
+    faThumbsDown = faThumbsDown;
+    faRedo = faRedo;
+
+    protected readonly IrisSender = IrisSender;
+
+    // State variables
+    messagesSubscription: Subscription;
+    stagesSubscription: Subscription;
+    errorSubscription: Subscription;
+    numNewMessageSubscription: Subscription;
+    rateLimitSubscription: Subscription;
+    activeStatusSubscription: Subscription;
+
+    messages: IrisMessage[] = [];
+    stages?: IrisStageDTO[] = [];
+    error?: IrisErrorMessageKey;
+    numNewMessages: number = 0;
+    rateLimitInfo: IrisRateLimitInformation;
+    active: boolean = true;
+
+    newMessageTextContent = '';
+    isLoading: boolean;
+    shouldAnimate: boolean = false;
+    hasActiveStage: boolean = false;
+
+    // User preferences
+    userAccepted: boolean;
+    isScrolledToBottom = true;
+    rows = 1;
+    resendAnimationActive: boolean;
+    public ButtonType = ButtonType;
+
+    @Input() chatService: IrisChatService;
+
+    @Input() fullSize: boolean | undefined;
+    @Input() showCloseButton: boolean = false;
+    @Output() fullSizeToggle = new EventEmitter<void>();
+    @Output() closeClicked = new EventEmitter<void>();
+
+    // ViewChilds
+    @ViewChild('messagesElement') messagesElement!: ElementRef;
+    @ViewChild('scrollArrow') scrollArrow!: ElementRef;
+    @ViewChild('messageTextarea') messageTextarea: ElementRef<HTMLTextAreaElement>;
+    @ViewChild('acceptButton') acceptButton: ElementRef<HTMLButtonElement>;
+
+    constructor(
+        protected accountService: AccountService,
+        protected modalService: NgbModal,
+        protected translateService: TranslateService,
+        protected statusService: IrisStatusService,
+    ) {}
+
+    ngOnInit() {
+        this.messagesSubscription = this.chatService.currentMessages().subscribe((messages) => {
+            if (messages.length !== this.messages?.length) {
+                this.scrollToBottom('auto');
+            }
+            this.messages = [...messages].reverse();
```

Comment:
(Not necessary to fix right now) if we immediately reverse the order of messages, maybe it would make more sense to allow setting the order when requesting the list from the server.